java/java.project.ui/src/org/netbeans/spi/java/project/support/ui/CreateClassAndCopy.java

```diff
@@ -0,0 +1,172 @@
+        catch (ClassNotFoundException x) {
+        } catch (ExceptionInInitializerError x) {
+        } catch (LinkageError x) {
+        } // exceptions from getMethod:
+        catch (SecurityException x) {
+            t = x;
+        } catch (NoSuchMethodException x) {
+            t = x;
+        } // exceptions from invoke
+        catch (IllegalAccessException x) {
+            t = x;
+        } catch (IllegalArgumentException x) {
+            t = x;
+        } catch (InvocationTargetException x) {
+            t = x;
+        }
+
+        if (t != null) {
+            Logger.getLogger(PasteAction.class.getName()).log(Level.WARNING, null, t);
+        }
+
+        return null;
+    }
+
+    @Override
+    public void actionPerformed(ActionEvent ev) {
+        try {
+            Clipboard c = Toolkit.getDefaultToolkit().getSystemClipboard();
+            if(!c.isDataFlavorAvailable(DataFlavor.stringFlavor))return;
+            boolean containsClass = false;
+            String data = (String) c.getData(DataFlavor.stringFlavor);
+            String className = "";
+            String uncommented = data.replaceAll("(?:/\\*(?:[^*]|(?:\\*+[^*/]))*\\*+/)|(?://.*)","");// NOI18N
+            uncommented = uncommented.trim();
+            StringTokenizer st = new StringTokenizer(uncommented, " ");// NOI18N
+            if (st.hasMoreTokens() && st.nextToken().equals("public") && st.hasMoreTokens() && st.nextToken().equals("class") && st.hasMoreTokens()) {// NOI18N
+                String cname = st.nextToken().trim();
+                if (cname.charAt(cname.length() - 1) == '{') {
+                    cname = cname.substring(0, cname.length() - 1);
+                }
+                if (SourceVersion.isIdentifier(cname)) {
+                    containsClass = true;
+                    className = cname;
+                }
+            }
+            if (!containsClass) {
+                JOptionPane.showMessageDialog(null, "Code not valid to create class");
+                return;
+            }
+            Set<FileObject> files = this.context.files();
+            if (files.size() != 1) {
+                return;
+            }
+            String path = files.iterator().next().getPath();
+            String packageName = "";
+            File f = new File(path + "\\" + className + ".java");// NOI18N
+            if (f.exists()) {
+                JOptionPane.showMessageDialog(null, "Can not create class already present");
+                return;
+            }
+            if (!f.createNewFile()) {
+                JOptionPane.showMessageDialog(null, "Can not create file");
+                return;
+            }
+            ExplorerManager explorerManager = findExplorerManager();
+            if (explorerManager.getSelectedNodes().length != 1) {
+                return;
+            }
+            Node selectedNode = explorerManager.getSelectedNodes()[0];
+            while (!selectedNode.getName().equals("${src.dir}")) {// NOI18N
```

Comment:
This feels really weird - I don't think anyone ever said anything about the names of the Nodes. Suggest to use ClassPath.getClassPath(SOURCE, target).getResourceName(target) (handling failures along the way).